include/core/Layer.hpp

```diff
@@ -20,9 +20,9 @@ namespace ngen
         public:
 
         #ifdef NGEN_MPI_ACTIVE
-            typedef hy_features::HY_Features_MPI feature_type;
+            using feature_type = hy_features::HY_Features_MPI;
         #else
-            typedef hy_features::HY_Features feature_type;
+            typedef feature_type = hy_features::HY_Features;
```

Comment:
Trying to work out the rebase, and when I get to this commit (about), this fails on GCC8, saying `feature_type` does not define a type... when I switch this to `using` instead of `typedef`, it seems to compile. Is there a particular reason the above is `using` and this is `typedef`? Or is it just an oversight? If the latter I can amend it in that commit. If the difference is deliberate, we can figure out what's going on after the rebase.